부산대 BE_오선재 Step1 - 엔티티 매핑

5seonjae opened the pull request:
### 📋 구현 내역

DB Migration
- `members`, `products`, `wish_items`, `approved_products` 테이블 DDL 스크립트 수정

Entity 클래스
- `Member`, `Product`, `ApprovedProduct`, `WishItem` 엔티티 정의 및 매핑
- 양방향 연관관계(@OneToMany, @ManyToOne), 제약조건(nullable, unique) 적용

Repository 인터페이스
- `MemberRepository`, `ProductRepository`, `ApprovedProductRepository`, `WishRepository` 생성
- 커스텀 메서드(existsByName, existsByEmail, findByEmail, findAllByMemberId, deleteByMemberIdAndProductId) 선언

설정 변경
- application.yml
    - MySQL/H2(test) 데이터소스 분리
    - spring.jpa.show-sql=true, hibernate.format_sql=true 설정
    - 테스트 프로파일에 ddl-auto: create 적용

Service 레이어 리팩토링
- 기존 JdbcClient* 구현체 → JPA 리포지토리 주입으로 전환

단위 테스트
- @DataJpaTest 기반 Repository 테스트 작성 (엔티티 매핑 및 커스텀 메서드 검증)

코드 정리
- 불필요해진 JdbcClient* 클래스 및 설정 제거

5seonjae committed:
fix: 폼 기반 찜 기능을 쿠키 인증으로 지원 및 템플릿 문법 통일

- TokenExtractor 컴포넌트 추가: Authorization 헤더 또는 “AUTH” 쿠키에서 JWT 추출
- LoginMemberArgumentResolver 개선: TokenExtractor 활용해 쿠키 파싱 지원
- 사용자 상품 목록 템플릿 수정: th:action="@{'/wishes/' + ${product.id}}" 문법 통일, “← 찜 목록으로” 버튼 추가
- 관리자 상품 목록 템플릿: 불필요한 찜 폼 제거

5seonjae committed:
docs: README.md 추가 및 Step1 엔티티 매핑 가이드 작성

- 프로젝트 목표 및 기능 요구사항 체크리스트(DDL 검토, Entity 매핑, Repository 정의, 설정, Service 리팩터링, 테스트) 작성
- MySQL/H2용 DDL 예시 스크립트(Products, ApprovedProducts, Members, WishItems) 포함
- JPA 마이그레이션 및 테스트 가이드라인 명시

5seonjae committed:
feat: Spring Data JPA 의존성 및 설정 추가

- `spring-boot-starter-data-jpa` 의존성 추가
- MySQL Dialect 설정 (`spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect`)
- Hibernate SQL 포맷 및 출력 설정 활성화
    - `spring.jpa.properties.hibernate.format_sql=true`
    - `spring.jpa.show-sql=true`

5seonjae committed:
feat: ApprovedProduct JPA 엔티티 및 카카오 상품 승인 검증 로직 통합

- ApprovedProduct 엔티티에 JPA 어노테이션(@Entity, @Table, @Id, @GeneratedValue) 추가
- ApprovedProductRepository 인터페이스를 JpaRepository<ApprovedProduct, Long>로 구현, existsByName 메서드 정의
- 서비스 레이어에 verifyKakaoNameIsApproved 메서드 적용: “카카오” 포함 상품명에 대해 데이터베이스 조회 기반 승인 여부 검증
- 기존 하드코딩 검증 로직을 JPA 조회로 전환

5seonjae committed:
test: 상품 수정 실패 테스트 가격을 양수로 변경

- `updateProduct_fail_imageUrlMissing` 테스트에서 DTO 가격을 `1500`으로 수정해 이미지 URL 검증 로직이 올바르게 실행되도록 개선

5seonjae committed:
refactor: Member·WishItem·Product 엔티티 및 레포지토리 JPA로 전환

- Member, WishItem, Product 클래스에 JPA 매핑 애노테이션(@Entity, @Table, @Id, @GeneratedValue, @OneToMany 등) 적용
- JpaRepository 기반 Repository 인터페이스로 변경 및 기존 커스텀 메서드(findAllByMemberId, deleteByMemberIdAndProductId) 유지
- 연관관계 매핑(FetchType.LAZY 등) 및 orphanRemoval 설정
- 기존 JDBCClient 로직 제거, 서비스 코드에서 JPA Repository 호출로 교체
- @DataJpaTest 기반 테스트 코드(MemberRepositoryTest, ProductRepositoryTest, WishRepositoryTest) 수정 및 추가
- application.properties H2 설정 업데이트
    - `spring.jpa.hibernate.ddl-auto`를 `create`로 변경하여 애플리케이션 시작 시 스키마 자동 생성
    - `spring.sql.init.mode`를 `never`로 설정해 `schema.sql` 초기화를 비활성화
    - `spring.jpa.database-platform`을 `org.hibernate.dialect.H2Dialect`로 지정

5seonjae committed:
chore: schema.sql 업데이트 – products.name 길이 조정 및 members.email 고유 제약 추가

- products.name 컬럼을 VARCHAR(15)로 수정
- members.email 컬럼에 UNIQUE 제약조건 추가

5seonjae committed:
docs: README.md 체크리스트 완료 및 DDL 섹션 개정

- 요구사항 체크리스트 항목 모두 체크 표시
- DDL 예시 스크립트 업데이트: products.name 길이 조정 및 members.email UNIQUE 제약 추가

jjongwa commented on README.md:
```diff
@@ -1 +1,77 @@
-# spring-gift-enhancement
\ No newline at end of file
+# spring-gift-enhancement
```
README 업데이트 잊지 않으셨군요! 👍 👍

jjongwa commented on src/main/java/gift/entity/WishItem.java:
```diff
@@ -1,49 +1,62 @@
+    @Column(nullable = false)
+    private int quantity;
+
+    public WishItem() {
```
모든 엔티티에 기본 생성자를 추가해 주셨군요! 
이러한 기본 생성자가 없으면 어떤 문제가 발생하나요?

jjongwa replied:
+현재 기본 생성자가 public으로 설정되어 있는데, 
그럼 다른 곳에서 마음껏 해당 생성자를 통해 필드가 아무것도 없는 객체가 만들어질 수 있겠네요! 

이를 어떻게 방지할 수 있을까요?

5seonjae replied:
>모든 엔티티에 기본 생성자를 추가해 주셨군요!
>이러한 기본 생성자가 없으면 어떤 문제가 발생하나요?

1. 기본 생성자 없이 발생하는 문제

JPA 구현체 ( Hibernate 등 ) 는 리플렉션을 통해 엔티티를 인스턴스화할 때 파라미터 없는 ( no‑args ) 생성자를 반드시 호출합니다.

JPA 스펙에서도 “엔티티 클래스는 public 또는 protected 레벨의 파라미터 없는 생성자를 가져야 한다”고 명시하고 있습니다.
> The entity class must have a no-arg constructor. The entity class may have other constructors as well. The no-arg constructor must be public or protected.

기본 생성자가 없다면 다음과 같은 예외가 발생할 수 있습니다.

현재 제 프로젝트의 Member 엔티티에 기본 생성자를 제거했다고 가정해보겠습니다.
<img width="241" height="58" alt="image" src="https://github.com/user-attachments/assets/bacffd78-fa74-4ba1-a587-f2e1d1c1ce19" />

Application 을 실행해보면, 정상적으로 실행됩니다.
<img width="1485" height="26" alt="image" src="https://github.com/user-attachments/assets/32bf8c9b-12e9-4754-a4ae-5b3308602dd1" />

하지만, 회원가입을 하고, 로그인을 하는 순간, 오류가 발생하면서 Whitelabel Error Page 가 보이게 됩니다.
<img width="666" height="163" alt="image" src="https://github.com/user-attachments/assets/842fd8c9-e2e3-4e75-aaa6-c14e11bb0bb2" />
<img width="1047" height="74" alt="image" src="https://github.com/user-attachments/assets/41bd0534-ce3d-46f4-805f-2e400f53e867" />

기본 생성자가 왜 필요한지도 같이 알아보았습니다.

<Hibernate ORM 참조 가이드>
>4.1.1. Implement a no-argument constructor

>Cat has a no-argument constructor. All persistent classes must have a default constructor (which can be non-public) so that Hibernate can instantiate them using Constructor.newInstance(). It is recommended that you have a default constructor with at least package visibility for runtime proxy generation in Hibernate. 

JPA 구현체 ( 예 : Hibernate ) 는 데이터베이스에서 조회한 결과를 엔티티 객체로 복원할 때 `Class.newInstance()` 나 유사한 리플렉션 메커니즘을 사용합니다.
이때 파라미터가 없는 생성자만 호출할 수 있으므로, 반드시 “인수가 없는 생성자” 가 있어야 합니다.

Java SE 17 Javadoc – Class.newInstance()
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Class.html#newInstance() 

Hibernate 같은 구현체는 연관 관계를 지연 로딩 ( lazy loading ) 하기 위해 엔티티의 서브클래스 ( 프록시 클래스 ) 를 생성합니다.
이 서브클래스는 부모 엔티티의 기본 생성자를 호출 ( super() ) 해서 초기화하기 때문에, 역시 파라미터 없는 생성자가 필요합니다.

---

>현재 기본 생성자가 public으로 설정되어 있는데,
>그럼 다른 곳에서 마음껏 해당 생성자를 통해 필드가 아무것도 없는 객체가 만들어질 수 있겠네요!
>이를 어떻게 방지할 수 있을까요?

2. `public` 기본 생성자 남용 방지

`public` 으로 두면 리뷰어님께서 말씀해주신 것과 같이 애플리케이션 내 어느 곳에서든 “필드가 초기화되지 않은” 엔티티가 생성될 수 있으므로, 외부 호출을 차단하기 위해 `protected` 접근 제어자로 변경해야 할 것 같습니다.
제가 이 부분을 간과한 것 같습니다.

`protected` 접근 제어자로 변경을 하면 JPA는 여전히 리플렉션으로 기본 생성자를 호출할 수 있고, 비즈니스 코드에서는 필수 필드를 갖춘 생성자만 사용하도록 강제해 객체의 무결성을 지킬 수 있습니다.

jjongwa replied:
👍 👏 💯

jjongwa commented on src/main/java/gift/entity/ApprovedProduct.java:
```diff
@@ -1,8 +1,21 @@
+@Table(name = "approved_products")
 public class ApprovedProduct {
 
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
+
+    @Column(nullable = false, unique = true)
     private String name;
 
     public ApprovedProduct() {
```
ApprovedProduct는 카카오 이름을 포함할 수 있는 Product를 위한 엔티티일까요?

jjongwa replied:
해당 엔티티는 WithItem에 포함되지 않는 것 같은데, 의도된 걸까요?

5seonjae replied:
> ApprovedProduct는 카카오 이름을 포함할 수 있는 Product를 위한 엔티티일까요?

1. ApprovedProduct의 역할

이 엔티티는 “관리자측에서 승인한 상품 목록 ( Whitelist )”을 관리하기 위한 마스터 테이블 역할입니다.
다시 말하자면, "카카오" 라는 문구가 들어있는 상품명에 대해서 등록/수정 시에 이 엔티티에 있는 상품명만 허용하는 역할이라고 볼 수 있습니다.

고유한 name 컬럼 하나만 갖고 있어서, 어떤 상품이 승인되었는지 여부만 판단할 수 있도록 설계했습니다.

---

> 해당 엔티티는 WithItem에 포함되지 않는 것 같은데, 의도된 걸까요?

2. WishItem 과의 분리 이유

현재 설계에서는 WishItem 이 문자열(name) 형태로만 상품명을 저장합니다.

이렇게 하면 승인된 이름이 바뀌더라도 WishItem의 기록(로그)에는 영향을 주지 않고, 단순히 “찜한 시점의 상품명”을 보존할 수 있다는 장점이 있습니다.

“카카오” 상품명에만 화이트리스트 적용한다는 부분에서 의도적으로 해당 엔티티를 WithItem 에 포함하지 않았습니다.

ApprovedProduct 테이블은 오직 “카카오”가 들어간 상품명을 등록/수정할 때만 허용 여부를 판별하기 위한 것이고, 일반 상품명 ( “카카오”가 아닌 경우 ) 은 아무 검증 없이 사용 가능해야 한다면, `@ManyToOne` 으로 무조건 매핑해 버릴 때, 일반 상품까지 외래 키 제약에 걸려버립니다.

따라서 WishItem 에는 문자열 ( productName ) 만 쓰되, “카카오”가 포함된 이름일 때는 서비스 계층 ( 또는 커스텀 Validator ) 에서 ApprovedProduct 조회를 통해 허용 여부를 체크하는 편이 더 적절하다고 생각했습니다.

jjongwa commented on src/main/java/gift/service/ProductService.java:
```diff
@@ -49,21 +49,21 @@ public Product updateProduct(long id, ProductUpdateRequestDto updateRequestDto)
     private void verifyKakaoNameIsApproved(String name) {
         if (name.contains("카카오")) {
-            if (!approvedRepository.isApproved(name)) {
+            if (!approvedRepository.existsByName(name)) {
```
아 aprroveProduct가 해당 Product 이름이 가능한 이름인지만 따로 모아둔 값이군요..!

음 제가 봤을 때 일반적으로 '카카오' 라는 이름을 사용해도 되는 물건들이 있고, 해당 물건들에만 이런 제한 조건이 해제되어야 할 것 같은데, 현재 구성된 로직에 따르면

1. '카카오 무드등' 이라는 네이밍을 ApprovedProduct에 저장
2. '카카오' 이름 허락을 받지 않은 '라인프렌즈 무드등' 이라는 Product 생성
3. 해당 Product의 이름을 '카카오 무드등' 으로 변경
4. 다른 상관 없는 물건들도 동일하게 이름을 '카카오 무드등'으로 변경

이러한 사태가 발생할 수 있을 것 같은데요..!
ApprovedProduct라는 테이블이 꼭 필요한지 다시 확인해 봐야 할 것 같네요.!

5seonjae replied:
날카로운 지적 감사합니다!

피드백을 보고 다시 한 번 제 로직을 살펴봤는데, 피드백에 있는 상황이 발생할 수 있다는 것을 인지하게 되었습니다.

ApprovedProduct 에서 이름만을 가지고 검증을 한다면 전혀 관련없는 물품에 대해서 문제가 발생할 수 있습니다.

그렇기에 이 방법을 개선하거나 다른 방법들을 찾아야 합니다.

이 방법을 개선하려고 곰곰히 생각을 해보았는데, 관련없는 물품에 대해 지금의 방식을 개선하는 것으로 해결이 되지 않을 것 같다는 생각이 들었습니다.

그래서 다른 방법을 생각해 보았는데, 이전에 이런 피드백을 받았던 것이 생각이 났습니다.
https://github.com/next-step/spring-gift-wishlist/pull/82#discussion_r2179851939

### ApprovedProduct 엔티티 제거 및 brand_requires_approval + approved 플래그 도입

해당 피드백에서 "Product 테이블에 requires_approval + approved 플래그 추가" 라는 대안에 대해 언급을 했었습니다.

이 방법을 사용하면 등록/수정 시에 "카카오" 라는 이름이 들어가기만 하면 일단 MD 의 허가가 있어야 하기 때문에, 관련없는 물품에 대해서도 문제가 발생하지 않을 것 같습니다.

그래서 이 방법으로 수정을 하려고 합니다.

### “import.sql에 MD 계정 추가 → MD 전용 승인 UI/REST API 구현 → Product 엔티티에 flag 추가 → 기존 ApprovedProduct 기능 제거

`import.sql` 에 MD 계정을 추가하고, MD 만 사용할 수 있는 승인 관련 페이지와 기능을 만들어서 "카카오" 이름이 들어간 물건에 대한 제약 조건을 구현할 생각입니다.
이에 따라, `ApprovedProduct` 엔티티와 관련 기능들 및 테이블을 제거하고 `Product` 엔티티에 "카카오" 이름 포함 여부를 판단하는 `requires_approval flag` 와 승인 여부를 판단하는 `approved` 플래그를 추가하겠습니다.

다만, 현 시점에는 구현 여력이 부족하여 바로 적용하기는 어려울 것 같습니다.

리뷰어님께서는 이 대안에 대해 어떻게 생각하시는지,
혹은 더 나은 대안이 있다면 편하게 공유 부탁드립니다!

jjongwa replied:
> 이에 따라, ApprovedProduct 엔티티와 관련 기능들 및 테이블을 제거하고 Product 엔티티에 "카카오" 이름 포함 여부를 판단하는 requires_approval flag 와 승인 여부를 판단하는 approved 플래그를 추가하겠습니다.

저도 괜찮다고 생각합니다! 여기서 조금 더 간단하게 가보자면 사실 정말 중요한건(물론 관점에 따라 다르겠지만) 해당 상품의 이름에 카카오가 포함되느냐 여부가 아닌 카카오 이름을 써도 되는 **관리자의 허락을 받은 상품인가** 일 수도 있겠네요

그럼 기존 Product 엔티티에 approved 필드 하나만 추가하면 되고, 상품 이름 수정 시에도 approved가 true인지에 따라 검증 로직을 진행하면 될 것 같아요!

jjongwa replied:
이전 PR에서의 답변과 이번 PR에서 제가 던진 질문에 대한 답변으로 봤을 때, 
우선 해당 기능을 이러한 방식으로 구현한 이유와, 구체적인 진행 방식까지 잘 설명해 주셨어요! 👍 

하지만 한가지 짚고 넘어가야 할 점은, 우선 코드로만 봤을 때는 쉽게 이해 + 설득되지 않는 구조일 수 있다는 거에요.
모수가 적긴 하지만 일단 지난 미션 리뷰어와 제가 거의 비슷한 질문을 한 걸로 봐서, 
해당 코드를 처음 접하는 사람 입장에서 충분히 해당 로직이 이해가 되는 흐름인가도 중요하게 봐야 할 요소라고 생각합니다!

그래서 본인이 코드를 짜다가 너무 딮하게 들어가는 것 같으면, 조금 거리를 두고 처음으로 돌아가서 
현재 요구사항을 최대한 간단히 충족할 수 있는 구성이 무엇일까? 를 생각하는 것도 추천드립니다~!

jjongwa commented on src/main/java/gift/entity/WishItem.java:
```diff
@@ -1,49 +1,62 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```
`@ManyToOne` 설정 👍 

fetchType과 optional 설정을 추가해 주었는데, 각 옵션은 어떤 기능을 하나요?

5seonjae replied:
### 1. `fetch = FetchType.LAZY`

- 의미
  - 연관된 대상 엔티티를 지연 로딩 ( lazy loading ) 하도록 힌트를 줍니다.

  - 실제로 해당 필드에 접근하는 시점에 SQL 을 실행해 DB에서 데이터를 가져옵니다.

엔티티 ( WishItem ) 를 조회할 때, 연관된 ApprovedProduct 객체는 즉시 함께 조회되지 않고, 실제로 사용할 때 ( 예 :  wishItem.getApprovedProduct() ) 지연 로딩됩니다.

- 장점
  - 불필요한 조인/쿼리를 줄여 성능을 개선할 수 있습니다.

### 2. `optional = false`

- 의미
  - 연관 관계가 필수 ( Non-Null ) 임을 지정합니다.

  - `optional=false` 로 설정하면 JPA 는 외래키 ( FK ) 컬럼에 `NOT NULL` 제약을 걸고, 런타임에 해당 관계가 항상 존재해야 함을 보장합니다.

---

이렇게 선언하면, 해당 필드를 조회할 때까지 연관된 Member ( 혹은 자식 쪽에서의 WishItem ) 로딩을 늦추면서, 관계가 없으면 애초에 저장조차 불가능하도록 보장할 수 있습니다.

jjongwa commented on src/main/resources/application.properties:
```diff
@@ -6,10 +6,15 @@ spring.application.name=spring-gift
 spring.datasource.url=jdbc:h2:mem:testdb;MODE=MySQL;DB_CLOSE_DELAY=-1
 spring.datasource.username=sa
 spring.datasource.password=
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
```
H2 Dialect 를 추가해 준 이유는 무엇인가요? 
해당 설정을 추가해 주지 않으면 어느 기준으로 sql 문법이 실행되게 되나요?

5seonjae replied:
좋은 질문 감사합니다!

> H2 Dialect 를 추가해 준 이유는 무엇인가요?

### 1. H2 전용 SQL 생성

- Hibernate 는 Dialect 에 따라 IDENTITY, LIMIT 절, IF EXISTS 같은 DB 별 고유 문법을 조합해서 SQL 을 만듭니다.

- H2 전용 기능을 완전히 활용하려면 `org.hibernate.dialect.H2Dialect` 를 사용해야 안전합니다.

### 2. 자동 탐지의 한계

- Spring Boot/Hibernate 는 기본적으로 DataSource 의 DatabaseMetaData ( 또는 JDBC URL/드라이버 클래스 ) 정보를 보고 방언을 자동 결정합니다.

- H2 드라이버 ( org.h2.Driver ) 를 쓰면 보통 H2Dialect 를, MySQL 드라이버를 쓰면 MySQLDialect 를 선택합니다.

`spring.jpa.database-platform` 설정을 제거해도 Spring Boot 의 자동 구성 덕분에 H2 를 사용할 때는 자동으로 H2Dialect 가 적용되어 정상 동작합니다. 다만, 특수한 상황에서 자동 탐지 로직이 엇갈릴 여지가 있어, 예기치 못한 SQL 오류를 방지하기 위해 명시적으로 H2Dialect 를 설정했습니다.

---

> 해당 설정을 추가해 주지 않으면 어느 기준으로 sql 문법이 실행되게 되나요?

Hibernate 는 `spring.jpa.database-platform` 을 지정하지 않으면 다음 기준으로 방언 ( Dialect ) 을 자동 결정합니다.

### 1. JDBC 메타데이터 조회

애플리케이션이 연결할 때 `Connection.getMetaData().getDatabaseProductName()` 등을 호출해 현재 데이터베이스 종류를 파악합니다.

- H2 드라이버면 “H2” → H2Dialect
- MySQL 드라이버면 “MySQL” → MySQLDialect
- PostgreSQL 드라이버면 “PostgreSQL” → PostgreSQLDialect

### 2. 자동 매핑 로직

위 메타데이터 값을 내부 매핑 테이블에 대입해 가장 적합한 Dialect 클래스를 선택합니다.

Spring Boot 의 자동 구성 ( Auto‑Configuration ) 이 이 과정을 래핑해 주므로, 보통 개발자가 신경 쓰지 않아도 알맞은 방언이 설정됩니다.

- 참고 자료

> The dialect to use is also automatically detected based on the current DataSource, but you can set spring.jpa.database yourself if you want to be explicit and bypass that check on startup.

https://docs.spring.io/spring-boot/docs/2.1.5.RELEASE/reference/html/howto-data-access.html#howto-jpa-dialect

### 3. Fallback: GenericDialect

메타데이터가 애매하거나 매핑 테이블에 해당 DB 가 없으면, Hibernate 는 ANSI SQL 만 최소한으로 지원하는 GenericDialect 를 사용합니다.

이 경우 일부 벤더 고유 문법 ( IDENTITY 전략, LIMIT/OFFSET 등 ) 이 제대로 동작하지 않을 수 있습니다.

- 참고 자료

Hibernate Core Reference Guide – Dialect Resolution §25.3
https://docs.redhat.com/en/documentation/red_hat_jboss_web_server/2.1/html/hibernate_core_reference_guide/dialect_resolution

GenericDialect 클래스 API 문서
https://docs.atlassian.com/hibernate2/2.1.8/api/net/sf/hibernate/dialect/GenericDialect.html

jjongwa replied:
잘 정리해 주셨어요!
현재는 따로 DB 연결까진 하지 않기 때문에 H2를 사용하지만, mysql 같은 DB를 연결할 때를 대비해 두 가지 모두 충족하는 문법을 사용하는 것도 고려 대상이 될 수 있겠네요!

+ 더 나아가면 실제와 test의 환경을 다르게 가져갈 수도 있구요!

jjongwa reviewed (changes requested):
1단계 잘 구현해 주셨어요!

몇가지 질문 남겼습니다! 코멘트에 대한 답변 & 반영 후 다시 요청 주세요~!

5seonjae committed:
chore: import.sql로 기본 데이터 시드 추가

- products 테이블에 초콜릿(₩1,000), 새우깡(₩1,500) 샘플 데이터 삽입
- approved_products 테이블에 카카오 프렌즈 필통·인형 샘플 데이터 삽입
- members 테이블에 관리자(5seonjae@gmail.com) 및 일반 회원(6seonjae@gmail.com) 계정 삽입

5seonjae committed:
refactor: 엔티티 기본 생성자 접근 제어자 public → protected

- Product, ApprovedProduct, Member, WishItem 클래스의 no-argument 생성자 접근 제한자를 public에서 protected로 변경

jjongwa reviewed (approved):
리뷰 반영 잘 해주셨어요~!
(질문들에 대해 깊게 공부한 흔적이 보여서 좋았습니다 👍 )

이번 단계는 여기서 마무리하고, 다음 단계에서 뵐께요~!
